src/RwaRegistry.sol

```diff
@@ -0,0 +1,563 @@
+     * @param name The unsupported component name.
+     */
+    error UnsupportedComponent(bytes32 name);
+
+    /**
+     * @notice Revert reason when trying to add an supported component more than once.
+     * @param name The component name.
+     */
+    error ComponentAlreadySupported(bytes32 name);
+
+    /**
+     * @notice Revert reason when trying to add components with mismatching params.
+     */
+    error MismatchingComponentParams();
+
+    /**
+     * @notice Revert reason when calling iterator methods with bad parameters.
+     */
+    error InvalidIteration();
+
+    /**
+     * @notice Revert reason when trying to get a component `name` which does not exist for the deal identified by `ilk`
+     * @param ilk The ilk name.
+     * @param name The unsupported component name.
+     */
+    error ComponentDoesNotExist(bytes32 ilk, bytes32 name);
+
+    /**
+     * @notice Only addresses with admin access can call methods with this modifier.
+     */
+    modifier auth() {
+        if (wards[msg.sender] != 1) {
+            revert Unauthorized();
+        }
+        _;
+    }
+
+    /**
+     * @notice The deployer of the contract gains admin access to it.
+     * @dev Adds the default supported component names to the registry.
+     */
+    constructor() {
+        _supportedComponents.add("urn");
+        emit AddSupportedComponent("urn");
+
+        _supportedComponents.add("liquidationOracle");
+        emit AddSupportedComponent("liquidationOracle");
+
+        _supportedComponents.add("outputConduit");
+        emit AddSupportedComponent("outputConduit");
+
+        _supportedComponents.add("inputConduit");
+        emit AddSupportedComponent("inputConduit");
+
+        _supportedComponents.add("jar");
+        emit AddSupportedComponent("jar");
+
+        _supportedComponents.add("jarInputConduit");
+        emit AddSupportedComponent("jarInputConduit");
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+              Authorization
+  //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /*//////////////////////////////////
+     Supported Components Management
+  //////////////////////////////////*/
+
+    /**
+     * @notice Adds a supported component name to the registry.
+     * @dev Adds a new type of component that should be supported.
+     * @param name The "pascalCased" name of the component.
+     */
+    function addSupportedComponent(bytes32 name) external auth {
```

Comment:
The current interfaces don't match with the ones from `ilk-registry` (see [here](https://www.diffchecker.com/4aw5UhFG/) for diffs).
Interfaces should be conformant with the Ilk Registry to ensure integrators could easily adopt the new ad-hoc registry for RWAs with a smooth transition from the previous ilk reference, making a good use of overloaded functions where new logic will be introduced (e.g. `iter` -> `list`).

Reply:
This is not meant to match the `ilk-registry` in any sense, as it's not a replacement for it.
RWAs deals are still stored in the `ilk-registry`, however the RWA Registry fills the gap where the `ilk-registry` falls short.